src/DynamoApplications/StartupUtils.cs

```diff
@@ -455,19 +484,23 @@ private static bool PreloadASM(string asmPath, out string geometryFactoryPath, o
             string commonDataFolder,
             string geometryFactoryPath,
             string preloaderLocation,
-            HostAnalyticsInfo info = new HostAnalyticsInfo())
+            HostAnalyticsInfo info = new HostAnalyticsInfo(),
+            bool isServiceMode = false)
         {
-            var config = new DynamoModel.DefaultStartConfiguration()
+            var config = new DynamoModel.DefaultStartConfiguration
             {
                 GeometryFactoryPath = geometryFactoryPath,
                 ProcessMode = CLImode ? TaskProcessMode.Synchronous : TaskProcessMode.Asynchronous,
                 HostAnalyticsInfo = info,
-                CLIMode = CLImode
+                CLIMode = CLImode,
+                AuthProvider = CLImode ? null : new Core.IDSDKManager(),
+                UpdateManager = CLImode ? null : OSHelper.IsWindows() ? InitializeUpdateManager() : null,
+                StartInTestMode = CLImode,
+                PathResolver = CLImode ? new CLIPathResolver(preloaderLocation, userDataFolder, commonDataFolder) as IPathResolver : new SandboxPathResolver(preloaderLocation) as IPathResolver,
+                // TODO: Dynamo 3.0
+                // Update Default start config to include IsServiceMode so DynamoModel can be initialize with that flag. For now, reuse the legacy Context property
```

Comment:
you can safely modify DefaultStartConfig to include IsServiceMode you just can't change the interface - so you can add a new interface.